session/session.go

```diff
@@ -2366,9 +2374,9 @@ func (s *session) Close() {
 	s.RollbackTxn(ctx)
 	if s.sessionVars != nil {
 		s.sessionVars.WithdrawAllPreparedStmt()
-		if s.sessionVars.StmtStats != nil {
-			s.sessionVars.StmtStats.SetFinished()
-		}
+	}
+	if s.stmtStats != nil {
+		s.stmtStats.SetFinished()
```

Comment:
How about add a `ReleaseResource` method in `sessionctx.Context`, since we need to release more resouce, not only this.   /cc @tiancaiamao PTAL

Reply:
I would suggest using a non-sys-pool session for such task, if it need `ReleaseSource`...